plugins/outputs/azure_monitor_logs/azure_monitor_logs.go

```diff
@@ -0,0 +1,253 @@
+package azure_monitor_logs
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+var sampleConfig = `
+  ## Customer ID (Workstation ID) and Key for Azure Log Analytics resource.
+  # customer_id = "<Workstation ID>"
+  # shared_key = "<Secret>"
+
+  ## Timeout for closing (default: 5s).
+  # timeout = "5s"
+
+  ## Table Namespace Prefix (default: "").
+  ## Namespace Prefix is used in "Log-Type" header
+  ## Prefix can only contain alphaNumeric characters
+  # namespace_prefix = ""
+`
+
+const (
+	baseURL                = "https://%s.ods.opinsights.azure.com/api/logs?api-version=2016-04-01"
+	contentType            = "application/json"
+	timeGeneratedFieldName = "DateTime"
+	namespacePrefixRegex   = "^[a-zA-Z0-9]*$"
+	httpMethod             = http.MethodPost
+
+	defaultClientTimeout = 5 * time.Second
+)
+
+// AzLogAnalytics contains information about a azure log analytics service metadata
+type AzLogAnalytics struct {
+	CustomerID string `toml:"customer_id"`
+	SharedKey  string `toml:"shared_key"`
+
+	NamespacePrefix string            `toml:"namespace_prefix"`
+	ClientTimeout   internal.Duration `toml:"timeout"`
+
+	serviceURL string
+	client     *http.Client
+}
+
+// Initializes the plugin
+func (a *AzLogAnalytics) Init() error {
+	if a.CustomerID == "" {
+		return fmt.Errorf("customer_id not configured")
+	}
+
+	if a.SharedKey == "" {
+		return fmt.Errorf("shared_key not configured")
+	}
+
+	if len(a.NamespacePrefix) > 25 {
+		return fmt.Errorf("namespace_prefix length is greater than 25 characters")
+	}
+
+	if a.ClientTimeout.Duration == 0 {
+		a.ClientTimeout.Duration = defaultClientTimeout
+	}
+
+	match, err := regexp.MatchString(namespacePrefixRegex, a.NamespacePrefix)
+	if err != nil {
+		return err
+	} else if !match {
+		return fmt.Errorf("namespace_prefix contains invalid characters")
+	}
```

Comment:
Please skip the `else` here as you return in the `if`
```suggestion
	if err != nil {
		return err
	} 
	if !match {
		return fmt.Errorf("namespace_prefix contains invalid characters")
	}
```

Reply:
Still valid.